flink-runtime/src/main/java/org/apache/flink/runtime/state/filesystem/FsCheckpointStreamFactory.java

```diff
@@ -374,7 +378,7 @@ public void close() {
 
         @Nullable
         @Override
-        public StreamStateHandle closeAndGetHandle() throws IOException {
+        public synchronized StreamStateHandle closeAndGetHandle() throws IOException {
```

Comment:
how about remove the `synchronized (this)` below, since you have synchronized here

Reply:
Good catch! Updated.